### ISSUE TITLE
Infinite loop in -[NSString stringByRemovingHTMLEntities]

### ISSUE THREAD

**Issue by bdaz** [alexaubry/HTMLString#31]
I'm using HTMLStrings 6.0.0. Looks like simply calling `-[NSString stringByRemovingHTMLEntities]` is causing an infinite loop. See attached.

<img width="1412" alt="Screen Shot 2020-09-28 at 12 17 56 PM" src="https://user-images.githubusercontent.com/1771280/94476038-c8bbe500-0184-11eb-9c46-84233e355fe5.png">

Example invocation context: 

```
NSString *const htmlEncodedString = @"Apple&#39;s Tim Cook on a &quot;giant leap&quot; in social progress;
NSString *const noEntities = [htmlEncodedString stringByRemovingHTMLEntities]; // Hangs
```

**Comment by JCSooHwanCho** [comment]
I made PR about it in #33

**Comment by alexaubry** [comment]
@bdaz Can you confirm this has been fixed in 6.0.1?

**Comment by bdaz** [comment]
@alexaubry Can confirm—fixed! Thank you.
